delivery-receipts-api/src/main/java/uk/gov/di/authentication/deliveryreceiptsapi/lambda/NotifyCallbackHandler.java

```diff
@@ -140,6 +129,47 @@ public APIGatewayProxyResponseEvent notifyCallbackRequestHandler(
         return generateEmptySuccessApiGatewayResponse();
     }
 
+    private void incrementCounters(
+            String sentMetricName,
+            Map<String, String> additionalContext,
+            NotifyDeliveryReceipt deliveryReceipt) {
+        var sentMetricsMap = new HashMap<String, String>();
+        sentMetricsMap.put("Environment", configurationService.getEnvironment());
+        sentMetricsMap.put("NotifyStatus", deliveryReceipt.status());
+        sentMetricsMap.putAll(additionalContext);
+
+        cloudwatchMetricsService.incrementCounter(sentMetricName, sentMetricsMap);
+
+        if (deliveryReceipt.status().equals("delivered")) {
+            sendDurationMetrics(deliveryReceipt);
+        }
+    }
+
+    private void sendDurationMetrics(NotifyDeliveryReceipt deliveryReceipt) {
+        if (Objects.isNull(deliveryReceipt.completedAt())
+                || Objects.isNull(deliveryReceipt.createdAt())) {
+            return;
+        }
```

Comment:
We could put a `hasBeenDelivered` (or appropriate name) on the record that returns `completedAt == null || createdAt == null` and use that here instead?

Reply:
Hmm, I'm going to leave this for now because I think it's more useful to see we've done null checks before doing the below lines, and I couldn't think of a good way of expressing the check here, but one for maybe some future refactoring